File: problems/1-easy/125-Valid-Palindrome/solution.py
def isPalindrome(s: str) -> bool:
    stripped_str = "".join([c for c in s.lower() if c.isalnum()])

    forward_index = 0
    backward_index = len(stripped_str) - 1

    while forward_index < backward_index:
        if stripped_str[forward_index] == stripped_str[backward_index]:
            forward_index += 1
            backward_index -= 1
        else:
            return False

    return True

File: problems/1-easy/125-Valid-Palindrome/solution.cpp
#include <string>
using namespace std;

bool isPalindrome(string s) {
    string filtered = "";
    for (char c : s) {
        if (isalnum(c)) {
            filtered.push_back(tolower(c));
        }
    }

    int start = 0;
    int end = filtered.size()-1;
    while (start < end) {
        if (filtered[start] != filtered[end]) {
            return false;
        }
        start++;
        end--;
    }
    return true;
}
